subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/internal/SourceFoldersCreator.java

```diff
@@ -217,13 +217,9 @@ private boolean containsAll(FileCollection first, FileCollection second) {
         return true;
     }
 
-    private void addSourceSetAttribute(SourceSet sourceSet, SourceFolder folder) {
-        // Using the test sources feature introduced in Eclipse Photon
-        String name = sourceSet.getName();
-        if (!SourceSet.MAIN_SOURCE_SET_NAME.equals(name)) {
-            if (SourceSet.TEST_SOURCE_SET_NAME.equals(name) || folder.getPath().toLowerCase().contains("test")) {
-                folder.getEntryAttributes().put(EclipsePluginConstants.TEST_SOURCES_ATTRIBUTE_KEY, EclipsePluginConstants.TEST_SOURCES_ATTRIBUTE_VALUE);
-            }
+    private void addSourceSetAttribute(SourceSet sourceSet, SourceFolder folder, Collection<SourceSet> testSourceSets) {
```

Comment:
Maybe `addTestSourceSetAttributeIfNeeded()`?  I'm a fan of very descriptive method names.

Reply:
Good idea. Method name adjusted.